File: Modules/RuleEngine/src/hap/ruleengine/state/BaseState.java
// Copyright (c) 2016 Per Malmberg
// Licensed under MIT, see LICENSE file.

package hap.ruleengine.state;

import hap.communication.Communicator;
import hap.communication.state.CommState;
import hap.message.cmd.Ping;
import hap.message.response.PingResponse;
import hap.ruleengine.RuleEngine;

public abstract class BaseState extends CommState
{
	public BaseState( Communicator com )
	{
		super( com );
	}

	@Override
	public void accept( Ping msg )
	{
		publish( new PingResponse( RuleEngine.class.getName() ) );
	}
}

File: Modules/RuleEngine/src/hap/ruleengine/state/BaseState.kt
// Copyright (c) 2016 Per Malmberg
// Licensed under MIT, see LICENSE file.

package hap.ruleengine.state

import hap.ruleengine.RuleEngine


abstract class BaseState(val engine : RuleEngine) : chainedfsm.EnterLeaveState()
{
	open fun tick() {}
}

